### DISCUSSION TITLE
Not able to install anchor.

### DISCUSSION THREAD

**Discussion by Aamirusmani1552** [otter-sec/anchor#3200] [Q&A] [answered]
Hey everyone, I am trying to install anchor on my pc. But got this error:

```bash
help: consider giving `items` an explicit type, where the placeholders `_` are specified
   |
83 |     let items: Box<_> = format_items
   |              ++++++++

   Compiling linux-raw-sys v0.4.8
   Compiling data-encoding v2.4.0
   Compiling jsonrpc-core v18.0.0
   Compiling asn1-rs-derive v0.4.0
For more information about this error, try `rustc --explain E0282`.
error: could not compile `time` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `anchor-cli v0.30.1 (https://github.com/coral-xyz/anchor?tag=v0.30.1#e6d7dafe)`, intermediate artifacts can be found at `/tmp/cargo-install1E8d2K`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
Error: Failed to install 0.30.1, is it a valid version?
Version 0.30.1 is not installed, no current version.
Error: You need to run 'avm install 0.30.1' to install it before using it.
```

It says we have to define the type explicitly. But how can I do so even if the code is not even mine. Or Maybe it's something else.
Need help on this.

I have tried to run the install command on windows, git bash and WSL. But result is still same. This is the command i am using:

```bash
cargo install --git https://github.com/coral-xyz/anchor avm --locked --force
```

**Comment by acheroncrypto** [comment, marked as answer]
Please see the "Note" section of https://github.com/coral-xyz/anchor/pull/3143.

Duplicate of https://github.com/coral-xyz/anchor/issues/3126, https://github.com/coral-xyz/anchor/issues/3158, https://github.com/coral-xyz/anchor/issues/3164, https://github.com/coral-xyz/anchor/issues/3169, https://github.com/coral-xyz/anchor/issues/3173, https://github.com/coral-xyz/anchor/issues/3198.
